R/bambu-extendAnnotations-utilityExtend.R

```diff
@@ -3,10 +3,11 @@
 #' @noRd
 isore.extendAnnotations <- function(combinedTranscripts, annotationGrangesList,
                                     remove.subsetTx = TRUE,
-                                    min.sampleNumber = 1, min.exonDistance = 35, min.exonOverlap = 10,
+                                    min.sampleNumber = 1, max.txNDR = 0.1, min.exonDistance = 35, min.exonOverlap = 10,
                                     min.primarySecondaryDist = 5, min.primarySecondaryDistStartEnd = 5, 
                                     prefix = "", verbose = FALSE){
   filterSet <- filterTranscriptsByRead(combinedTranscripts, min.sampleNumber)
+  filterSet <- filterTranscriptsByNDR(combinedTranscripts, filterSet, max.txNDR)
```

Comment:
suggestion: replace filterTranscriptsByRead with a function filterTranscripts() that does the following:

- first filter by min.sampleNumber
- then calculate NDR and add to combined transcripts (it might be very helpful to return this with the extended annotations in the final output)
- then filter by max.txNDR
- return combinedTranscripts object that is now filtered and which contains the NDR estimate

there might be better ways to do this, the main point would be to make this easier to read/maintain and to keep the NDR in the final output

Reply:
Resolved the first suggestion in 6f39f8936d29f8eb36ec48ea9397e502b27ee408
Outputing the NDR estimate here wouldn't do anything as isore.extendAnnotations outputs annotations (GRanges) and not the combinedTranscripts. I could change the output but I am not sure what would then use this downstream?

Reply:
Added the NDR to the annotations mcols in 82f29273d644bff24bcd5cd5bc5bf89db6c7d2cb